src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -227,27 +225,20 @@ def get_existing_version_info(self, version: str) -> Optional[dict]:
         It executes a 'show versions like ... in application package' query and returns the result as single row, if one exists.
         """
         with self.use_role(self.package_role):
-            show_obj_query = f"show versions like '{unquote_identifier(version)}' in application package {self.package_name}"
-
             try:
-                show_obj_cursor = self._execute_query(
-                    show_obj_query, cursor_class=DictCursor
+                version_obj = self.show_specific_object(
+                    "versions",
+                    version,
+                    name_col=VERSION_COL,
+                    in_clause=f"in application package {self.package_name}",
                 )
```

Comment:
nit: could be shortened to `return self.show_specific_object(...)`, no need to wait until after the except block

Reply:
You'd think so, but sadly the linter complains that there's no return (because the except block calls the generic error handler). I could add an explicit `return None` there, might feel cleaner?

Reply:
ah you're right about the except block, this probably works best then